packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts

```diff
@@ -1,21 +1,33 @@
 import { builtinModules } from "node:module";
 import nodePath from "node:path";
 import dedent from "ts-dedent";
-import { cloudflare, env, nodeless } from "unenv";
+import { cloudflare, defineEnv } from "unenv";
 import { getBasePath } from "../../paths";
 import type { Plugin, PluginBuild } from "esbuild";
 
 const REQUIRED_NODE_BUILT_IN_NAMESPACE = "node-built-in-modules";
 const REQUIRED_UNENV_ALIAS_NAMESPACE = "required-unenv-alias";
 
 export const nodejsHybridPlugin: () => Plugin = () => {
-	const { alias, inject, external } = env(nodeless, cloudflare);
+	// Get the resolved environment.
+	const { env } = defineEnv({
+		nodeCompat: true,
+		presets: [cloudflare],
+		resolve: true,
+	});
+	const { alias, inject, external } = env;
+	// Get the unresolved alias.
+	const unresolvedAlias = defineEnv({
+		nodeCompat: true,
+		presets: [cloudflare],
+		resolve: false,
+	}).env.alias;
```

Comment:
Interesting would you need it for longer them you think? We could also expose `resolveEnv()` from unenv to avoid defining twice. (but also it is so to call it twice haha!)

Reply:
Having a `resolveEnv()` or `env.unresolved` would make things slightly nicer here but I'm fine with calling it twice.

Reply:
Happy to go with the current approach. But it seems to me, what would be nice is if `unenv` exposed something like `env.resolve(path)` function that can be used to get the absolute path of any import specifier.
Then there would be no need to hold two distinct alias objects.

Reply:
Another good question.

We are iterating with Pooya to find the best way to do this. We might revisit this code as we progress.